ui/guest/src/react/ExperienceBuilder.tsx

```diff
@@ -687,10 +685,12 @@ function ExperienceBuilderInternal(props: InternalGuestProps) {
                         : sxStylesConfig.zoneMarker.selectModeHighlight,
                       { clone: true }
                     ),
-                    hasValidations || isLocked || isExternallyModified
+                    hasValidations || isLocked || isExternallyModified || !isEditable
                       ? hasFailedRequired || isExternallyModified
                         ? sxStylesConfig.zoneMarker.errorHighlight
-                        : sxStylesConfig.zoneMarker.warnHighlight
+                        : isLocked
+                          ? sxStylesConfig.zoneMarker.warnHighlight
+                          : sxStylesConfig.zoneMarker.disabledHighlight
```

Comment:
Fix logic. 
This is changing to show the warn highlight only when locked and there are/may-be other cases (e.g. warning level requirement)